packages/server/src/standalone/index.ts

```diff
@@ -0,0 +1,149 @@
+import { json } from 'body-parser';
+import cors from 'cors';
+import express from 'express';
+import http from 'http';
+import type { AddressInfo, ListenOptions } from 'net';
+import { format as urlFormat } from 'url';
+import { ApolloServer } from '../ApolloServer';
+import { ExpressContext, expressMiddleware } from '../express';
+import type { BaseContext, ContextFunction } from '../externalTypes';
+import type { WithRequired } from '../types';
+
+interface HTTPServerOptions<TContext extends BaseContext> {
+  context?: ContextFunction<[ExpressContext], TContext>;
+}
+
+export function httpServer(
+  server: ApolloServer<BaseContext>,
+  options?: {},
+): HTTPApolloServer<BaseContext>;
+export function httpServer<TContext extends BaseContext>(
+  server: ApolloServer<TContext>,
+  options: WithRequired<HTTPServerOptions<TContext>, 'context'>,
+): HTTPApolloServer<TContext>;
+export function httpServer<TContext extends BaseContext>(
+  server: ApolloServer<TContext>,
+  options?: HTTPServerOptions<TContext>,
+): HTTPApolloServer<TContext> {
+  const context = options?.context ?? (async () => ({} as TContext));
+  return new HTTPApolloServer<TContext>(server, { context });
+}
+
+class HTTPApolloServer<TContext extends BaseContext> {
+  private app: express.Express = express();
+  private httpServer: http.Server = http.createServer(this.app);
+
+  constructor(
+    private apolloServer: ApolloServer<TContext>,
+    private options: WithRequired<HTTPServerOptions<TContext>, 'context'>,
+  ) {}
+
+  async listen(
+    listenOptions: ListenOptions = { port: 4000 },
+  ): Promise<{ url: string }> {
+    await this.apolloServer.start();
+    this.app.use(
+      cors<cors.CorsRequest>(),
```

Comment:
fwiw my recollection is we don't need the `<>` here; it was a specific weird issue with inference with the variant of `.use` that took a path.

Reply:
[f6b44fd](https://github.com/apollographql/apollo-server/pull/6420/commits/f6b44fd4e533e3abad21f0bd4029b816888a9bf3)